File: examples/record/prepare/prepare.ts
const { RecordClient, Bloock } = require('@bloock/sdk');
const fs = require('fs');

try {
  let recordClient = new RecordClient();
  let record = await recordClient.fromString('Hello world').build();

  // we can get the hash of the record
  let hash = await record.getHash();
  console.log(hash);

  // build a record from an existing record
  let record2 = await recordClient.fromRecord(record).build();

  // we can read a file as an array of bytes
  const file = fs.readFileSync('sample.pdf');
  let record3 = await recordClient.fromFile(file).build();
} catch (e) {
  console.log(e);
}

File: examples/record/prepare/prepare.java
import java.nio.file.Files;
import java.nio.file.Paths;

import com.bloock.sdk.Bloock;
import com.bloock.sdk.client.RecordClient;
import com.bloock.sdk.entity.Record;

class Test {
  public static void main(String[] args) {
    try {
      RecordClient recordClient = new RecordClient();

      Record record = recordClient.fromString("Hello world").build();

      // we can get the hash of the record
      String hash = record.getHash();
      System.out.println(hash);

      // build a record from an existing record
      Record record2 = recordClient.fromRecord(record).build();

      // we can read a file as an array of bytes
      byte[] file = Files.readAllBytes(Paths.get("sample.pdf"));
      Record record3 = recordClient.fromFile(file).build();

    } catch (Exception e) {
      System.out.println(e);
    }
  }
}
